org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/EventDispatcher.java

```diff
@@ -155,11 +155,16 @@ private void dispatch(EventSet eventSet) {
 				}
 				vote = true;
 				try {
-					resume = listener.handleEvent(event, fTarget, !resume, eventSet) && resume;
+					try {
+						resume = listener.handleEvent(event, fTarget, !resume, eventSet) && resume;
+					} finally {
+						if (!fTarget.isDisconnected()) {
+							enableGCForExceptionEvent(event);
+						}
+					}
+				} catch (VMDisconnectedException e) {
```

Comment:
I'm not sure if we should do more on `VMDisconnectedException` here... Maybe `shutdown()` and `break`?

Reply:
I see in `org.eclipse.jdt.internal.debug.core.logicalstructures.JavaLogicalStructure.releaseValue(IValue)` we don't care too.

Other way to deal with that is `org.eclipse.jdt.internal.debug.core.model.JDIDebugElement.logError(Exception)`
can you please check if this approach could work in `enableGCForExceptionEvent()` too? I mean, only log if the target is still available.

Reply:
I'm not sure about this. We are in the loop that notifies everything about events, including the debug target. It might be possible the debug target has not been notified yet that the VM was disconnected. But please check the latest patch set.